metricflow-semantics/metricflow_semantics/experimental/semantic_graph/builder/entity_join_subgraph.py

```diff
@@ -0,0 +1,165 @@
+from __future__ import annotations
+
+import logging
+
+from typing_extensions import override
+
+from metricflow_semantics.experimental.dsi.join_lookup import SemanticModelJoinLookup
+from metricflow_semantics.experimental.dsi.manifest_object_lookup import ManifestObjectLookup
+from metricflow_semantics.experimental.dsi.model_object_lookup import (
+    ModelObjectLookup,
+)
+from metricflow_semantics.experimental.semantic_graph.builder.subgraph_generator import (
+    SemanticSubgraphGenerator,
+)
+from metricflow_semantics.experimental.semantic_graph.edges.sg_edges import (
+    EntityRelationshipEdge,
+    JoinToModelEdge,
+)
+from metricflow_semantics.experimental.semantic_graph.model_id import SemanticModelId
+from metricflow_semantics.experimental.semantic_graph.nodes.entity_nodes import (
+    ConfiguredEntityNode,
+    JoinedModelNode,
+    LocalModelNode,
+)
+from metricflow_semantics.experimental.semantic_graph.sg_interfaces import (
+    SemanticGraphEdge,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class EntityJoinSubgraphGenerator(SemanticSubgraphGenerator):
+    """Generator for the subgraph that represents the joins that are possible between semantic models.
+
+    Following the current query interface, joins between semantic models are modeled as a path from a model node
+    to a configured-entity node, and then to another model node.
```

Comment:
In this context, does the configured entity node represent the entity configured in the left semantic model or the right one? Or is it more a representation of all entities with the same name?